core/src/main/java/org/infinispan/encoding/impl/StorageConfigurationManager.java

```diff
@@ -109,6 +108,8 @@ private MediaType getStorageMediaType(Configuration configuration, boolean embed
       if (internalCache) return MediaType.APPLICATION_OBJECT;
 
       if (embeddedMode) {
+         boolean heap = configuration.memory().storage() == StorageType.OBJECT ||
+                        configuration.memory().storage() == StorageType.HEAP;
```

Comment:
I reckon ```configuration.memory().storage() == StorageType.HEAP``` is the only check you need, since the config will translate from the old attributes. Do you have a failing test for this JIRA @danberindei or was it just something you noticed?

Reply:
I don't have a test, I just noticed it while testing the anchored keys module, which at the time replaced cache values with `Address` instances and then treated them as protostream-encoded `byte[]`.

And it looks like I got the problem wrong: my test was using the deprecated `MemoryStorageConfigurationBuilder.storageType()`, and `HEAP` actually stopped working when I switched it to the new `MemoryConfigurationBuilder.storage()`. That's because they use 2 independent attributes, and `MemoryConfigurationBuilder` doesn't just translate OBJECT->HEAP when the attribute is set, it also translates back STORAGE=HEAP -> STORAGE_TYPE=OBJECT in `read()`. IMO the less magic in the configuration (or anywhere else, for that matter) the better, so personally I would remove the translation OBJECT->HEAP and back, and allow both OBJECT and HEAP here.

Reply:
> my test was using the deprecated MemoryStorageConfigurationBuilder.storageType(), and HEAP actually stopped working

Still don't get it. Were you using ```storageType(HEAP)``` and then moved to ```storage(HEAP)``` ?

Reply:
Yeah, looking at ```ScaleUpWithoutStateTransferTest```, it is the former

Reply:
I finally wrote a comprehensive test, and the only broken config is

```
      configurationBuilder.memory().storageType(StorageType.HEAP);
      assertStorageMediaTypes(configurationBuilder, StorageType.HEAP, StorageType.HEAP,
                              MediaType.APPLICATION_PROTOSTREAM);
```

Reply:
ok, just waiting for CI